### PR TITLE
Improve latency correction

### LINKED ISSUE #39: Xivalexander Performance Fluctuation
I downloaded the latest version of xivalexander and when I turn it on the performance fluctuates. I hit a dummy for 12 minutes and sometimes I can double weave sometimes I can't. At least when I press two ogcds in quick succession on my screen it looks like I didn't double weave it. But when I track it I am apparently able to double weave properly, but my screen shows that my skill didnt go off and I have to press my skills twice sometimes. 

Edit: I just tested it again with a MCH and when I used to be able to weave a ricochette/gauss round between heat blast, now I can't

### PR DESCRIPTION
**Pull request by jjolano**
This change to the latency correction feature will better handle discrepancies in ping to server response time by estimating the latency value based on server response time statistics. Fake ping VPN users will benefit from this change, but having real ping will still be better. For all users, consistency is improved.

Latency estimation will occur depending on these factors:
- Stability of server response time (deviation value)
- Difference of measured ping vs server response time after correcting any sharp spikes

For testing on my end, I artificially increased my server RTT by leaving a download running in the background - meanwhile my ping remains the same. This is most likely a result of packet congestion, which ping does not reflect accurately. By estimating latency to the server based on server response time statistics, packet congestion is factored out.

Penalty calculation (and `BaseLatencyPenalty`) has been removed since it does not improve consistency as originally intended.

TL;DR: Rewrote latency correction so the program does not disadvantage fake VPN users anymore. Also improves consistency and attempts to factor out packet congestion by scaling latency estimation towards server response times.

Fixes #39

### CODE REVIEW

**Comment by jjolano** [comment]
@Soreepeong while I think the estimation is pretty good, it is limited in testing for different environments, so this PR might need extra testing - maybe a beta?

**Comment by Soreepeong** [comment]
I think this goes into the territory of "why not?", and it does seem better when I tried this on 280ms ping.

* [EU.txt](https://github.com/Soreepeong/XivAlexander/files/6433080/EU.txt)
* [NA.txt](https://github.com/Soreepeong/XivAlexander/files/6433081/NA.txt)

**Comment by Soreepeong** [comment]
Now that I'm checking it again: do you remember why did you put `% originalWaitTime` here? @jjolano 

https://github.com/Soreepeong/XivAlexander/pull/52/commits/e6163deb2b21816d89bdd28259d314e4636fc405#diff-361589af9f42d0bfc4ecfca3a094a5e76d7bf8793983baec295d820781586f56R201

This doesn't seem right,

**Comment by jjolano** [comment]
> Now that I'm checking it again: do you remember why did you put `% originalWaitTime` here? @jjolano
> 
> [e6163de#diff-361589af9f42d0bfc4ecfca3a094a5e76d7bf8793983baec295d820781586f56R201](https://github.com/Soreepeong/XivAlexander/commit/e6163deb2b21816d89bdd28259d314e4636fc405#diff-361589af9f42d0bfc4ecfca3a094a5e76d7bf8793983baec295d820781586f56R201)
> 
> This doesn't seem right,

I believe that was a "band-aid" fix for server response times exceeding the original animation lock time (when trying to figure out why response times were unexplainably high in a previous issue). It probably has no effect being there, so it can safely be reverted.
